### ISSUE TITLE
`make install` does not install headers when shared library is disabled

### ISSUE THREAD

**Issue by brandond** [WasmEdge/WasmEdge#2256]
### Current State

`make install` does not install headers

### Expected

Header are available

## Environment

- Hardware Architecture: x86\_64
- Operating system: Alpine

```
- C++ Compiler version: clang 15.0.7
- CMake version: 3.24.3
- CMake flags: '-DCMAKE_BUILD_TYPE=Release' '-DWASMEDGE_BUILD_STATIC_LIB=ON' '-DWASMEDGE_BUILD_SHARED_LIB=OFF' '-DWASMEDGE_BUILD_AOT_RUNTIME=OFF' '-DWASMEDGE_BUILD_TOOLS=OFF' -Wno-dev
- Boost version: 1.80
```

## Steps to Reproduce

1. `cmake` as described above
2. `make install`
3.  note that only the static lib is installed; no headers are available. The install target output ends with:
```
make  -f CMakeFiles/Makefile2 preinstall
make[1]: Entering directory '/drone/src/WasmEdge-0.11.2'
make[1]: Nothing to be done for 'preinstall'.
make[1]: Leaving directory '/drone/src/WasmEdge-0.11.2'
Install the project...
/usr/bin/cmake -P cmake_install.cmake
-- Install configuration: "Release"
-- Installing: /usr/local/lib/libwasmedge.a
```

**Comment by brandond** [comment]
Ah, I see that the headers are inside the `if(WASMEDGE_BUILD_SHARED_LIB)` section of CMakeList.txt, so I guess I need to build the shared lib if I want the headers?

**Comment by q82419** [comment]
Hi @brandond ,

This issue has been resolved in the PR #2170.
Please try the `0.12.0-alpha.1` first or wait the `0.12.0` official release.
Thanks.

**Comment by hydai** [comment]
This is fixed by #2170 and already shipped after the 0.12.0 release.
